keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Hamiltonian.java

```diff
@@ -22,36 +26,50 @@
  * The No-U-Turn Sampler: Adaptively Setting Path Lengths in Hamiltonian Monte Carlo
  * https://arxiv.org/pdf/1111.4246.pdf
  */
-public class Hamiltonian {
+@Builder
+public class Hamiltonian implements PosteriorSamplingAlgorithm {
 
-    private Hamiltonian() {
+    public static Hamiltonian withDefaultConfig() {
+        return withDefaultConfig(KeanuRandom.getDefaultRandom());
     }
 
+    public static Hamiltonian withDefaultConfig(KeanuRandom random) {
+        return Hamiltonian.builder()
+            .random(random)
+            .leapFrogCount(20)
+            .stepSize(0.1)
```

Comment:
Not needed but I figured if someone decompiled the class they'd get the lombok generated stuff and it might not be clear what the defaults are unless this method specified them. It should at least be the same defaults as the builder uses, which I've updated it to make sure happens.